storage/operation/pebbleimpl/iterator.go

```diff
@@ -0,0 +1,68 @@
+package pebbleimpl
+
+import (
+	"fmt"
+
+	"github.com/cockroachdb/pebble"
+
+	"github.com/onflow/flow-go/storage"
+)
+
+type pebbleIterator struct {
+	*pebble.Iterator
+}
+
+var _ storage.Iterator = (*pebbleIterator)(nil)
+
+func newPebbleIterator(reader pebble.Reader, startPrefix, endPrefix []byte, ops storage.IteratorOption) (*pebbleIterator, error) {
+	lowerBound, upperBound, hasUpperBound := storage.StartEndPrefixToLowerUpperBound(startPrefix, endPrefix)
+
+	options := pebble.IterOptions{
+		LowerBound: lowerBound,
+		UpperBound: upperBound,
+	}
+
+	// setting UpperBound to nil if there is no upper bound
+	if !hasUpperBound {
+		options.UpperBound = nil
+	}
+
+	iter, err := reader.NewIter(&options)
+	if err != nil {
+		return nil, fmt.Errorf("can not create iterator: %w", err)
+	}
+
+	return &pebbleIterator{
+		iter,
+	}, nil
+}
+
+// IterItem returns the current key-value pair, or nil if done.
+func (i *pebbleIterator) IterItem() storage.IterItem {
+	return pebbleIterItem{i.Iterator}
+}
+
+// Next seeks to the smallest key greater than or equal to the given key.
+func (i *pebbleIterator) Next() {
+	i.Iterator.Next()
+}
```

Comment:
redundant method

Reply:
It's not redundant. The pebble's Next returns a bool, which indicates if next item is valid, and our Iterator doesn't return it. The reason for that is for badger, we added additional check in the Valid, so we should not just use the original Valid value from the database library, but should always use the customized `Valid` method implemented by the storage implementation.

Reply:
ah, sorry, didn't notice!